Assets/IntegrationTests/TestShopifyProducts.cs

```diff
@@ -84,9 +84,45 @@ public class TestShopifyProducts : MonoBehaviour {
                 "Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0Lzk4OTUyODE0NzU=", "Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0Lzk4OTUyODQyOTE="
             );
 
-            yield return Utils.GetWaitQuery ();
+            yield return waiter;
+
+            Assert.IsTrue(waiter.IsStopped, Utils.MaxQueryMessage);
```

Comment:
I still think it should be the job of the waiter to fail the test if it times out, which would remove this boilerplate

Reply:
Yeah maybe. Let's leave for now and refactor later when there's time.